src/nvim/popupmenu.c

```diff
@@ -812,9 +969,24 @@ static bool pum_set_selected(int n, int repeat)
 
           curbuf->b_changed = false;
           curbuf->b_p_ma = false;
+          if (pum_selected != prev_selected) {
+            curwin->w_topline = 1;
+          } else if (curwin->w_topline > curbuf->b_ml.ml_line_count) {
+            curwin->w_topline = curbuf->b_ml.ml_line_count;
+          }
           curwin->w_cursor.lnum = 1;
           curwin->w_cursor.col = 0;
 
+          if (use_float) {
+            if (curwin->w_p_wrap) {
+              lnum += plines_win(curwin, lnum, true);
+            }
+            pum_adjust_float_position(curwin, lnum, max_info_width);
+            if (win_valid(curwin_save)) {
+              redraw_later(curwin_save, UPD_SOME_VALID);
+            }
```

Comment:
Is this needed? If old-curwin had changes it is expected to already redraw itself.

Reply:
not sure. before removed also work. don't why bram add this.